File: src/main/java/de/check24/ui/pages/home/HomePage.java
package de.check24.ui.pages.home;

import de.check24.ui.driver.Driver;
import de.check24.ui.pages.base.BasePage;

import static de.check24.ui.driver.Driver.*;
import static de.check24.ui.pages.home.HomeText.*;


public class HomePage extends BasePage {
    private final String FOOTER_COMPANY_LINE = "//div[@class='c24-footer-company-line']";
    private final String LABEL_AMNELDEN = "//div[@class='c24-customer-icon c24-customer-icon-lo c24-icon']";
    private final String LABEL_AKTIVITATEN = "//a[@class='c24-activities-icon c24-header-hover']";
    private final String QUICK_CHIPS_LINK_HOTEL = "//a[@data-identifier='hotel']";
    private final String SEARCH_HEADER = "//input[@id='c24-search-header']";

    public String getFooterCompanyLine() {
        return getText(FOOTER_COMPANY_LINE);
    }

    public boolean isLabelAnmeldenDisplayed() {
        return isElementDisplayedWithWait(LABEL_AMNELDEN, 5);
    }

    public boolean isLabelAktivitätenDisplayed() {
        return isElementDisplayedWithWait(LABEL_AKTIVITATEN, 5);
    }

    public void clickQuickChipsLinkHotel() {
        Driver.click(QUICK_CHIPS_LINK_HOTEL);
    }

    public boolean isPlaceholderValueEqualsToExpected() {
        return isAttributeValueEqualsToExpected(SEARCH_HEADER,"placeholder",SEARCH_HEADER_PLACEHOLDER_EXPECTED_TEXT);
    }
}

File: js/test/pageobjects/home.page.js
import { $ } from '@wdio/globals'
import BasePage from './base.page.js'

class HomePage extends BasePage {

    #ACTIVITIES = "//div[@class='c24-activities-label c24-header-icon-label']";
    #HEADER_SEARCH = "//input[@id='c24-search-header']";
    #QUICK_CHIPS_LINK_HOTEL = "//a[@data-identifier='hotel']";
<<<<<<< HEAD
=======
    #ANMELDEN_LABEL = "//div[contains(@class, 'c24-customer-icon-lo')]";

    get anmeldenLabel() {
        return $(this.#ANMELDEN_LABEL);
    }
    
    get quickChipsLinkHotel() {
        return $(this.#QUICK_CHIPS_LINK_HOTEL);
    }

    async clickQuickChipsLinkHotel() {
        await this.quickChipsLinkHotel.click();
    }
>>>>>>> js

    get headerSearch () {
        return $(this.#HEADER_SEARCH);
    }

    get btnSubmit() {
        return $('button[type="submit"]');
    }

    get activities() {
        return $(this.#ACTIVITIES);
    }

    async clickQuickChipsLinkHotel() {
        await this.click(this.#QUICK_CHIPS_LINK_HOTEL);
    }
}

export default new HomePage();
